### PR TITLE
clusterctl 1.8.0

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by adilGhaffarDev** [comment]
@chenrui333  kindly check what is the issue here?

**Comment by iMichka** [comment]
The text has to be changed in the "assert" in the test. It should now be `Error: unable to load in-cluster configuration`. Can you update the pull request accordingly?

File: Formula/c/clusterctl.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class Clusterctl < Formula
   desc "Home for the Cluster Management API work, a subproject of sig-cluster-lifecycle"
   homepage "https://cluster-api.sigs.k8s.io"
-  url "https://github.com/kubernetes-sigs/cluster-api/archive/refs/tags/v1.7.4.tar.gz"
-  sha256 "c0acb1626a3948107db8131aff01900f9f097fc0b854ef5792fcd91dd324e4f6"
+  url "https://github.com/kubernetes-sigs/cluster-api.git",
+      tag:      "v1.8.0",
+      revision: "69bbf3530777a81867328c9379b41f814b008cdb"
```
Comment (chenrui333) [inline Formula/c/clusterctl.rb:6]: why do we do this?

Comment (adilGhaffarDev) [reply Formula/c/clusterctl.rb:6]: according to this guide: https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md#to-submit-a-version-upgrade-for-the-foo-formula 
It should be allowed to use tag and revision. But I can change it to sha256 if that is better

Comment (adilGhaffarDev) [reply Formula/c/clusterctl.rb]: changed it to url and sha

**Review by bevanjkay — APPROVED** [review]

File: Formula/c/clusterctl.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/c/clusterctl.rb]: yeah, we'd prefer tarball/sha whenever possible, but we did run into some go build with helm before https://github.com/Homebrew/homebrew-core/pull/177094

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**